test/ui/PipelineContextTest.ts

```diff
@@ -54,18 +54,75 @@ export function PipelineContextText() {
         it('should return the context value', () => {
           expect(test.cmp.getContextValue('qwerty')).toBe('azerty');
         });
+
+        it('should support setting a new context', () => {
+          test.cmp.setContext({ abc: 'def', hij: ['klm', 'nop'] });
+          expect(test.cmp.getContextKeys()).toEqual(jasmine.arrayContaining(['abc', 'hij']));
+        });
+
+        it('should support setting a single context value', () => {
```

Comment:
I think the test description should be tweaked to include the array case. Maybe something like:
`should support setting a single context value and an array of context values`.

Alternative may be to break the test into two.